appinventor/blocklyeditor/src/field_lexical_variable.js

```diff
@@ -297,10 +297,42 @@ Blockly.FieldLexicalVariable.getLexicalNamesInScope = function (block) {
                      && (parent.getInputTargetBlock('DO') == child)) {// Only DO is in scope, not other inputs!
               var rangeName = parent.getFieldValue('VAR');
               rememberName(rangeName, rangeNames, Blockly.loopRangeParameterPrefix);
-          } else if ((parent.type === "local_declaration_expression"
-                        && parent.getInputTargetBlock('RETURN') == child) // only body is in scope of names
-                      || (parent.type === "local_declaration_statement"
-                           && parent.getInputTargetBlock('STACK') == child) // only body is in scope of names
+          } else if ( (parent.type === "lists_map")
+                     && (parent.getInputTargetBlock('TO') == child)) {// Only DO is in scope, not other inputs!
+              var loopName = parent.getFieldValue('VAR');
```

Comment:
The indentation here and in all the `else if` clauses below should be 2 spaces.

Reply:
Fixed